### PR TITLE
Introduced RequestCredentials class

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1329

### Change description ###
This PR goes before https://github.com/hmcts/bulk-scan-orchestrator/pull/1063 to prevent quality gateway failure due to exceeding max number of method parameters. This has been discussed with @lgonczar as a preferable approach rather than suppressing check for the number of parameters.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackServiceTest.java [hunk 1]
```diff
@@ -135,7 +135,12 @@ void process_should_process_supplementary_evidence_with_ocr() {
         // given
         given(exceptionRecordValidator.mandatoryPrerequisites(any())).willReturn(Validation.valid(null));
         given(ccdCaseUpdater.updateCase(
-            exceptionRecord, configItem, true, IDAM_TOKEN, USER_ID, EXISTING_CASE_ID, EXISTING_CASE_TYPE
+            eq(exceptionRecord),
+            eq(configItem),
+            eq(true),
+            any(RequestCredentials.class),
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackServiceTest.java:141]: so now the tokens don't matter?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/RequestCredentials.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+public class RequestCredentials {
+    final String idamToken;
+    String s2sToken;
+    final String userId;
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/RequestCredentials.java:6]: it would be nice if all were final

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java [hunk 3]
```diff
@@ -73,11 +72,12 @@ public ProcessResult updateCase(
 
         try {
             String s2sToken = s2sTokenGenerator.generate();
+            requestCredentials.setS2sToken(s2sToken);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java:75]: why?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackServiceTest.java [hunk 4]
```diff
@@ -192,7 +209,14 @@ void process_should_not_update_case_if_error_occurs() {
         assertThat(res.getLeft().getErrors()).isEqualTo(asList("error1"));
 
         verify(ccdCaseUpdater)
-            .updateCase(exceptionRecord, configItem, true, IDAM_TOKEN, USER_ID, EXISTING_CASE_ID, EXISTING_CASE_TYPE);
+            .updateCase(
+                eq(exceptionRecord),
+                eq(configItem),
+                eq(true),
+                any(RequestCredentials.class),
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackServiceTest.java:216]: we should verify those too

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackService.java [hunk 5]
```diff
@@ -62,7 +62,7 @@
 @Service
 public class AttachToCaseCallbackService {
 
-    public static final String PAYMENT_ERROR_MSG = "Payment references cannot be processed. Please try again later";
+    private static final String PAYMENT_ERROR_MSG = "Payment references cannot be processed. Please try again later";
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackService.java:65]: is this change related?

**Comment by apereverzin** [comment]
https://github.com/hmcts/bulk-scan-orchestrator/pull/1068
